src/main/java/com/cloudant/client/internal/views/PageMetadata.java

```diff
@@ -30,23 +32,129 @@
     /**
      * The parameters needed to request the page described by this metadata
      */
-    ViewQueryParameters<K, V> pageRequestParameters;
+    final ViewQueryParameters<K, V> pageRequestParameters;
 
     /**
      * The page number of the page to retrieve
      */
-    long pageNumber;
+    final long pageNumber;
     /**
      * Indicate the paging direction for this metadata
      */
-    PagingDirection direction;
+    final PagingDirection direction;
 
     enum PagingDirection {
+        @SerializedName("F")
         FORWARD,
+        @SerializedName("B")
         BACKWARD
     }
 
-    PageMetadata(PagingDirection direction) {
+    /**
+     * Construct metadata that encapsulates the request query for an additional page.
+     *
+     * @param direction             the direction of the page
+     * @param pageNumber            the number of the page
+     * @param pageRequestParameters the query parameters to request the page
+     */
+    PageMetadata(PagingDirection direction, long pageNumber, ViewQueryParameters<K, V>
+            pageRequestParameters) {
+        // Set the direction and page number.
         this.direction = direction;
+        this.pageNumber = pageNumber;
+        this.pageRequestParameters = pageRequestParameters;
+    }
+
+    /**
+     * Generate query parameters for a forward page with the specified start key.
+     *
+     * @param initialQueryParameters page 1 query parameters
+     * @param startkey               the startkey for the forward page
+     * @param startkey_docid         the doc id for the startkey (in case of duplicate keys)
+     * @param <K>                    the view key type
+     * @param <V>                    the view value type
+     * @return the query parameters for the forward page
+     */
+    static <K, V> ViewQueryParameters<K, V> forwardPaginationQueryParameters
+    (ViewQueryParameters<K, V> initialQueryParameters, K startkey, String startkey_docid) {
+        ViewQueryParameters<K, V> pageParameters = new ViewQueryParameters<K, V>
+                (initialQueryParameters);
+        //set the start parameters
+        pageParameters.setStartKey(startkey);
+        pageParameters.setStartKeyDocId(startkey_docid);
+
+        //stay the same for forward
+        pageParameters.inclusive_end = initialQueryParameters.inclusive_end;
+        pageParameters.descending = initialQueryParameters.descending;
+        if (initialQueryParameters.endkey != null) {
+            pageParameters.setEndKey(initialQueryParameters.endkey);
+        }
+        pageParameters.setEndKeyDocId(initialQueryParameters.endkey_docid);
+
+        //all other parameters stay the same
+        copyNonPagingParameters(initialQueryParameters, pageParameters);
```

Comment:
Do you mean by making ViewQueryParameters `Cloneable`?
Otherwise isn't any method or code to set all the parameters as vulnerable to the addition of new parameters as the non-paging one (although I agree a method that copies everything is probably easier to maintain).